ql/instruments/makeois.cpp

```diff
@@ -97,20 +102,24 @@ namespace QuantLib {
         Schedule fixedSchedule(startDate, endDate,
                                Period(fixedPaymentFrequency),
                                fixedCalendar_,
-                               ModifiedFollowing,
-                               ModifiedFollowing,
+                               fixedConvention_,
+                               fixedTerminationDateConvention_,
                                fixedRule,
-                               usedEndOfMonth);
+                               fixedEndOfMonth);
         ext::optional<Schedule> overnightSchedule;
-        if (fixedPaymentFrequency != overnightPaymentFrequency || fixedRule != overnightRule ||
-            fixedCalendar_ != overnightCalendar_) {
+        if (fixedPaymentFrequency != overnightPaymentFrequency ||
+            fixedCalendar_ != overnightCalendar_ ||
+            fixedConvention_ != overnightConvention_ ||
+            fixedTerminationDateConvention_ != overnightTerminationDateConvention_ ||
+            fixedRule != overnightRule ||
+            fixedEndOfMonth != overnightEndOfMonth) {
```

Comment:
This `if` is becoming a bit too much—I'd create `overnightSchedule` anyway instead of having it optional.  If it's the same as the fixed schedule, so be it.

Reply:
I was trying to avoid unnecessary work for the common case (creating a schedule can be expensive). Can change to always create two.

Reply:
Done.